### PR TITLE
Change Redis to Valkey in log messages

### PR DESCRIPTION
**Pull request by 9bany**
Log messages containing "Redis" in some files are changed.

Add macro SERVER_TITLE defined to "Valkey" (uppercase V) is introduced and used in log messages, so at least it will be easy to patch this definition to get Redis or any other name in the logs instead of Valkey.

Change "Redis" in some log messages to use %s and SERVER_TITLE.

This is a partial implementation of https://github.com/valkey-io/valkey/issues/207

### CODE REVIEW

**Comment by 9bany** [comment]
a smaller PR  😄 @zuiderkwast

**Comment by PingXie** [comment]
Let's use the indirection (via SERVER_NAME) set up by #47 so we can easily build a binary that logs in the exact same way.

**Comment by 9bany** [comment]
@PingXie i see, let me make some changes

**Review by Shivshankar-Reddy — APPROVED** [review]

**Review by zuiderkwast — CHANGES_REQUESTED** [review]
We'll probably want this to be controlled by a compatibility config, to be as compatible as possible with redis in terms of script parsing the logs, etc. We'll not do this in our first release anyway.

Please leave this PR open for discussion so we can decide later.

**Comment by 9bany** [comment]
Related: https://github.com/valkey-io/valkey/issues/43?notification_referrer_id=NT_kwDOAy_BhbQxMDAxNDY5ODYyMTo1MzQ2MTM4MQ#issuecomment-2040691610

**Comment by zuiderkwast** [comment]
> Please leave this PR open for discussion so we can decide later.

Closed the pr? Don't want to leave it open?

**Comment by 9bany** [comment]
Sorry, i forget to open a issue  
i created one https://github.com/valkey-io/valkey/issues/240

**Comment by zuiderkwast** [comment]
I'm going to open this again.

**Comment by 9bany** [comment]
What is the problem needed to change in this PR @zuiderkwast  ?

**Comment by zuiderkwast** [comment]
> What is the problem needed to change in this PR @zuiderkwast ?

Nothing. I just want it to be open and waiting. :)

I wrote earlier "Please leave this PR open for discussion so we can decide later."

I want this PR but I am not a dictator of valkey. The other maintainers need to decide about it too. But right now we are busy with our first release (7.2.4) and this change is better for the next release (8.0).

**Review by zuiderkwast — APPROVED** [review]

**Review by zuiderkwast — CHANGES_REQUESTED** [review]
Don't merge. See #207.

File: src/version.h
Comment (madolson) [inline src/version.h:6]: Was the reason to make this SERVER_TITLE to make it independent from SERVER_NAME so we could add the compatibility switch later?

Comment (zuiderkwast) [reply src/version.h:6]: It's "Valkey" vs "valkey". It was discussed elsewhere (some clicks away).

@0del please update the PR title and write a description about the change, including what SERVER_TITLE is.

Comment (zuiderkwast) [reply src/version.h:6]: And yes, also for the compat switch.

Comment (madolson) [reply src/version.h:6]: Maybe call it SERVER_NAME_LOWER then? At first glance I didn't notice the difference

Comment (zuiderkwast) [reply src/version.h:6]: TITLE is in "title case", i.e. first letter capitalized.

(It was my suggestion.)

File: src/acl.c
Comment (zuiderkwast) [inline src/acl.c:2575]: The purpose of having %s here is to be keep to the log messages exactly as they were in Redis, because someone may search for exactly this string in the logs. If we want to support that, we must also keep "redis.conf" instead of "valkey.conf". Otherwise the string is still changed.

Personally, I think it makes no have the special compatibility mode for a log message like this one.

@PingXie FYI

File: src/version.h
Comment (zuiderkwast) [inline src/version.h:6]: Let's add a comment to clarify the difference between these two macros.

```suggestion
/* These macros are used where the server name is printed in logs and replies.
 * Note the difference in the first letter "V" vs "v". SERVER_TITLE is used in
 * readable text like log messages and SERVER_NAME is used in INFO fields and
 * similar. */
#define SERVER_NAME "valkey"
#define SERVER_TITLE "Valkey"
```

**Review by zuiderkwast — APPROVED** [review]
I'm fine with these changes. We've decided that log messages don't need to be identical to Redis log messages (except the markers for the crash report).

File: src/version.h
Comment (9bany) [reply src/version.h:6]: Thanks

File: src/acl.c
Comment (zuiderkwast) [reply src/acl.c:2575]: It's a little bit strange that with the macro, it's possible for a user to patch the macro, like `#define SERVER_TITLE "Redis"`, to make this message say "Redis" but it will still contain "valkey.conf" so the message is still not identical to in Redis. But we decided that we don't need to be 100% compatible in these log messages, I'm going to ignore that. We need to finish this rebranding some time. :)

**Pull request merged**
